### PR TITLE
consolidate env handling into pkg/env

### PR DESCRIPTION
**Pull request by vrothberg**
Env-variable related code is scattered across several packages making it
hard to maintain and extend.  Consolidate the code into a new pkg/env
package.

Signed-off-by: Valentin Rothberg <rothberg@redhat.com>
Signed-off-by: Matthew Heon <matthew.heon@pm.me>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: pkg/env/env.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+)
+
+// DefaultEnvVariables set $PATH, $TERM and $container.
+var DefaultEnvVariables = map[string]string{
+	"PATH":      "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+	"TERM":      "xterm",
+	"container": "podman",
+}
+
+const whiteSpaces = " \t"
```
Comment (TomSweeneyRedHat) [inline pkg/env/env.go]: Should there be a space in front of the tab?  I'm not sure.

Comment (rhatdan) [reply pkg/env/env.go]: Yes, this is either a space or a tab.

Comment (TomSweeneyRedHat) [reply pkg/env/env.go]: OK, I'm not sure I was clear, should this be `"\t"` or `"{tab or space}\t"` ?

Comment (vrothberg) [reply pkg/env/env.go]: It's used in `strings.TrimLeft` and meant to trim _tabs and whitespaces_, so we're good. `TrimLeft` interprets the string as a set of characters (in contrast to a string/sequence) for trimming.

**Comment by vrothberg** [comment]
Ready to go. @TomSweeneyRedHat @baude @rhatdan @giuseppe PTAL

File: pkg/env/env.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// DefaultEnvVariables set $PATH, $TERM and $container.
+var DefaultEnvVariables = map[string]string{
+	"PATH":      "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+	"TERM":      "xterm",
+	"container": "podman",
```
Comment (rhatdan) [inline pkg/env/env.go]: How about 
"container": path.Base(os.Executable())
This would make this package able to be used by non podman.

**Pull request merged**